deeplearning/ml4pl/graphs/programl.py

```diff
@@ -219,21 +232,67 @@ def ProgramGraphToNetworkX(proto: programl_pb2) -> nx.MultiDiGraph:
     g.graph[
       "data_flow_positive_node_count"
     ] = proto.data_flow_positive_node_count
+  
+  # Graph-level profile info
+  if proto.llvm_profile.HasField("num_functions"):
+    g.graph["llvm_profile_num_functions"] = proto.llvm_profile.num_functions
+  if proto.llvm_profile.HasField("max_function_count"):
+    g.graph["llvm_profile_max_function_count"] = proto.llvm_profile.max_function_count
+  if proto.llvm_profile.HasField("num_counts"):
+    g.graph["llvm_profile_num_counts"] = proto.llvm_profile.num_counts
+  if proto.llvm_profile.HasField("total_count"):
+    g.graph["llvm_profile_total_count"] = proto.llvm_profile.total_count
+  if proto.llvm_profile.HasField("max_count"):
+    g.graph["llvm_profile_max_count"] = proto.llvm_profile.max_count
+  if proto.llvm_profile.HasField("max_internal_count"):
+    g.graph["llvm_profile_max_internal_count"] = proto.llvm_profile.max_internal_count
 
   # Build the nodes.
   for i, node in enumerate(proto.node):
-    g.add_node(
-      i,
-      type=node.type,
-      text=node.text,
-      preprocessed_text=node.preprocessed_text,
-      function=(
-        proto.function[node.function].name
-        if node.HasField("function")
-        else None
-      ),
-      x=list(node.x),
-      y=list(node.y),
+    if not (node.HasField("llvm_profile_true_weight") or \
+            node.HasField("llvm_profile_false_weight") or \
+            node.HasField("llvm_profile_total_weight")):
+      g.add_node(
+        i,
+        type=node.type,
+        text=node.text,
+        preprocessed_text=node.preprocessed_text,
+        function=(
+          proto.function[node.function].name
+          if node.HasField("function")
+          else None
+        ),
+        x=list(node.x),
+        y=list(node.y),
+      )
+    else: # add profile info to node creation
```

Comment:
I see what you're doing here, but I don't like having the duplicate `if`/`else` code - it means there are twice as many places to check when changing the node representation.

Might I suggest a refactor in which we build a dictionary of all the node properties, which we can then pass to `g.add_node()`. This will make it easier to extend the node representaiton in future with new optional attributes. Like:
```py
node_data = {
  "type": node.type,
  "text": node.text,
  # ... repeat for all "essential" node properties
}
# repeat for all non-essential node properties:
if node.HasField("llvm_profile_true_weight"):
  node_data["llvm_profile_true_weight"] = node.llvm_profile_true_weight
# construct the actual graph node:
g.add_node(i, **node_data)
```

Reply:
mh! much better.